tests/unittests/tests-gnrc_ipv6_nib/tests-gnrc_ipv6_nib-internal.c

```diff
@@ -1449,6 +1449,65 @@ static void test_nib_offl_alloc__success_overwrite_unspecified(void)
     TEST_ASSERT(ipv6_addr_equal(&next_hop, &dst1->next_hop->ipv6));
 }
 
+/*
+ * Creates an off-link entry (to a next hop) and an on-link entry on the same interface.
+ * Then proceeds to delete the off-link entries to this next hop
+ * by only comparing the next hop, not checking the _PFX_ON_LINK flag.
+ *
+ * Expected results: Only off-link entries are deleted.
+ * On-link entries on the same interface are unaffected by the deletion.
+ */
+static void test_nib_offl_alloc__next_hop_indicates_whether_onl(void)
+{
+    _nib_offl_entry_t *dst1;
+    void *state = NULL;
+    static const ipv6_addr_t next_hop = { .u64 = { { .u8 = LINK_LOCAL_PREFIX },
+                                                 { .u64 = TEST_UINT64 } } };
+    static const ipv6_addr_t pfx = { .u64 = { { .u8 = GLOBAL_PREFIX } } };
+
+    /*are in practice different prefixes actually*/
+    static const ipv6_addr_t *onl_pfx = &pfx;
+    static const ipv6_addr_t *offl_pfx = &pfx;
+
+    /* Add off-link entry */
+    TEST_ASSERT_NOT_NULL((dst1 = _nib_offl_alloc(&next_hop, IFACE, offl_pfx,
```

Comment:
Sorry, when looking at the test I just wondered whether this should actually be a call to `_nib_ft_add()`?
Like a adding a routing rule.

Reply:
Yes, makes a bit more sense. And the test coverage remains the same. :+1: